flink-queryable-state/flink-queryable-state-runtime/src/test/java/org/apache/flink/queryablestate/network/KvStateServerHandlerTest.java

```diff
@@ -544,6 +552,7 @@ public void testIncomingBufferIsRecycled() throws Exception {
 
 		channel.writeInbound(unexpected);
 		assertEquals("Buffer not recycled", 0L, unexpected.refCnt());
+		channel.finishAndReleaseAll();
```

Comment:
I am curious of why we add this action only for this test, and actually the allocated buffers in this unit test are already released before.

Reply:
Here we are sending an expected message, i.e., a bad request, which will be answered with a request failure response:
https://github.com/apache/flink/blob/0619a5b3e000855f00d11e71ae2f1b9deef05c98/flink-queryable-state/flink-queryable-state-client-java/src/main/java/org/apache/flink/queryablestate/network/AbstractServerHandler.java#L262-L264

However, we are not reading the response and releasing the `ByteBuf` instance. You can set a breakpoint and debug into the `EmbeddedChannel` and check that there are indeed unread outbound messages.